File: crest3d/core/components/DebugComponent.cpp
#include "DebugComponent.h"
#include "Common.h"
void DebugComponent::start() {
	std::cout << "DebugComponent: Override function executing on start()" << std::endl;
}

void DebugComponent::earlyUpdate(float deltaTime) {
	std::cout << "DebugComponent: Override function executing on eaylyUpdate()" << std::endl;
}

void DebugComponent::update(float deltaTime) {
	std::cout << "DebugComponent: Override function executing on update()" << std::endl;
}

void DebugComponent::ui(float deltaTime) {
	std::cout << "DebugComponent: Override function executing on ui" << std::endl;
}

void DebugComponent::debugPrintSomething(const char* thing)
{
	std::cout << thing << std::endl;
}

File: crest3d/res/scripts/DebugComponent.lua
local class = require 'classy'

DebugComponent = class("DebugComponent")

function DebugComponent:initialize()
    
end

function DebugComponent:init()
    Component.init()
end

function DebugComponent:start()
    Component.start()
end

function DebugComponent:earlyUpdate()
    Component.earlyUpdate()
end

function DebugComponent:update()
    Component.update()
end

function DebugComponent:fixedUpdate()
    Component.fixedUpdate()
end

function DebugComponent:render()
    Component.render()
end

function DebugComponent:ui()
    Component.ui()
end

function DebugComponent:doSomething()
    print("Throw the jew down the well")
end

local MyComponent = DebugComponent:new()